src/systems/system.C

```diff
@@ -1109,29 +1110,54 @@ unsigned int System::add_variable (const std::string & var,
 
   else
     {
+      // Note that we only compare to the most recently added variable group
+      // so that we only create variable groups based on variables that are
+      // added consecutively to the system.
```

Comment:
I was surprised that we only check _variable_groups.back() here, rather than all of _variable_groups. I guess it's so that we only create variable groups based on variables that are added consecutively? I added a comment to explain that, but let me know if it's correct or not.

Reply:
OK, thanks for confirming.